interop-tests/src/compat.rs

```diff
@@ -0,0 +1,256 @@
+use libp2p::core::muxing::StreamMuxerBox;
+use libp2p::core::transport::Boxed;
+use libp2p::PeerId;
+
+// Native re-exports
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::build_transport;
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::init_logger;
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::sleep;
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::swarm_builder;
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::RedisClient;
+#[cfg(not(target_arch = "wasm32"))]
+pub use std::time::Instant;
+
+// Wasm re-exports
+#[cfg(target_arch = "wasm32")]
+pub use wasm::build_transport;
+#[cfg(target_arch = "wasm32")]
+pub use wasm::init_logger;
+#[cfg(target_arch = "wasm32")]
+pub use wasm::sleep;
+#[cfg(target_arch = "wasm32")]
+pub use wasm::swarm_builder;
+#[cfg(target_arch = "wasm32")]
+pub use wasm::RedisClient;
+#[cfg(target_arch = "wasm32")]
+pub use wasm_timer::Instant;
```

Comment:
I'm not really convinced that this will simplify a lot, because here the 'complexity' means just more `cfg` calls on those reexports where here it will mean duplication of the test function, but I can give this a try